components/forms/Form/Form.test.js

```diff
@@ -121,7 +121,9 @@ describe("Form Functionality", () => {
     render(<Form formRecord={formRecord} language="en" t={(key) => key} />);
     expect(screen.getByRole("button", { type: "submit" })).toBeInTheDocument();
 
-    await user.click(screen.getByRole("button", { type: "submit" }));
+    await waitFor(async () => {
+      await user.click(screen.getByRole("button", { type: "submit" }));
```

Comment:
Thanks for this catch @bryan-robitaille. While using `waitFor` did get rid of the warning, it definitely could have caused other issues. I was able to get rid of the warning by using `fireEvent.click()` instead of `userEvent.click()`.

While there are differences between `fireEvent` and `userEvent`, I think it should be suitable for this test.